include/deal.II/lac/block_matrix_base.h

```diff
@@ -1495,7 +1495,12 @@ template <typename MatrixType>
 inline
 BlockMatrixBase<MatrixType>::~BlockMatrixBase ()
 {
-  clear ();
+  try
+    {
+      clear ();
+    }
+  catch (...)
+    {}
```

Comment:
```
try {Assert/AssertThrow/AssertNothrow ( false, ExcMessage("3"));}
  catch (...) {AssertNothrow(false, ExcMessage("4"));}
```

results for `dealii::deal_II_exceptions::disable_abort_on_exception();` in

-  AssertThrow:    DEAL::Exception: ExcMessage("4"), no abort
-  Assert:              DEAL::Exception: ExcMessage("4"), no abort
-  AssertNothrow: DEAL::Exception: ExcMessage("3"), no abort

and without `dealii::deal_II_exceptions::disable_abort_on_exception();` in
-  AssertThrow:    Abort due to `AssertNothrow(false, ExcMessage("4"))`
-  Assert:              Abort due to `AssertNothrow(false, ExcMessage("3"))`
-  AssertNothrow: Abort due to `AssertNothrow(false, ExcMessage("3"))`

This behavior does not change when wrapping everything in another `try{...} catch(...) {}` block.
In the end, this is exactly what I would expect. Does this answer your question?